realm/src/androidTest/java/io/realm/RealmResultsTest.java

```diff
@@ -550,6 +553,27 @@ public void testSortWithTwoLanguages() {
         }
     }
 
+    public void testSortByChildObject() {
+        testRealm.beginTransaction();
+        Owner owner = testRealm.createObject(Owner.class);
+        owner.setName("owner");
+        Cat cat = testRealm.createObject(Cat.class);
+        cat.setName("cat");
+        owner.setCat(cat);
+        testRealm.commitTransaction();
+
+        RealmQuery<Owner> query = testRealm.where(Owner.class);
+        RealmResults<Owner> owners = query.findAll();
+
+        try {
+            owners.sort("cat.name");
+        } catch (IllegalArgumentException e) {
+            return; // Not supported yet
+        }
+
+        fail("Sorting by child object properties should result in a IllegalArgumentException");
```

Comment:
If you put this fail() right after owners.sort() you won't need the return. And it's how we check for exceptions in the other tests